apps/yggdrasil_web/web/controllers/character_controller.ex

```diff
@@ -0,0 +1,58 @@
+defmodule YggdrasilWeb.CharacterController do
+  use YggdrasilWeb.Web, :controller
+  use Guardian.Phoenix.Controller
+
+  alias Yggdrasil.Character
+
+  def index(conn, _params, user, _claims) do
+    user_id = user.id
```

Comment:
@copenhas Ok, so with this controller I had a to make a decision either assume the user in the token was the driving this or let the api provide the user_id. 

If the api provides the api, then I obviously need to verify the user_id against the token to ensure that the user is requesting just there stuff.

The corner case here is the Admin, who may either want to see just there characters or all characters. So this would basically require permissions at that point.

With permissions, the api can then make a decision on what to return based on it.

However there are subtle differences here, like creating a user for example would require a user_id to create it with, and for a normal user this would be their id and should never be anyone else's, for and admin could be either. Same for deletes, the character to be deleted for a normal user needs to be the user id in that toke not anyone else's.

So my question is would this case be special enough that it would be simpler possibly more secure to have two routes.
```
/characters
/admin/characters
```
The first one is what we have now and basically all operations the server fills in the user_id on the server based on the token. One this eliminates the checks to ensure any passed in value matches the token, and two simplifies things.

The second one when we get to that point could offer a more expansive set of things you can do that we would not normally want. In which case we can one guard that route with permissions, and two once entered we know that this route is special so the entire controller can make the assumption that this is an admin vs one controller handling both.

Thoughts? The current implementation assumes that all actions on the controller are for one user, and is scoped to the token's user_id. It was the safer of the two approaches without a lot of extra code and me being a littler conservative at the moment.